src/potential/gravfft.c

```diff
@@ -1178,28 +1176,58 @@ GMT_LOCAL void gravfft_load_from_top_grid (struct GMT_CTRL *GMT, struct GMT_GRID
 		f *= i;	/* n! */
 	p = n - 1.0;
 
-	twopi = 2 * M_PI;
 	D = (YOUNGS_MODULUS * Ctrl->T.te * Ctrl->T.te * Ctrl->T.te) / (12.0 * (1.0 - POISSONS_RATIO * POISSONS_RATIO));
-	alfa = pow(twopi,4.) * D / (NORMAL_GRAVITY * Ctrl->T.rho_mc);
+	alfa = pow(TWO_PI,4.) * D / (NORMAL_GRAVITY * Ctrl->T.rho_mc);
 	raised[0] = 0.0f;		raised[1] = 0.0f;
 
 	for (k = 0; k < Grid->header->size; k+= 2) {
-		mk = gmt_fft_get_wave (k, K) / twopi;
+		mk = gmt_fft_get_wave (k, K) / TWO_PI;
 		if (p == 0.0)
 			t = 1.0;
 		else if (p == 1.0)
 			t = mk;
 		else
 			t = pow (mk, p);
 		earth_curvature = (sphericity) ? (2 * earth_rad * mk) / (4 * M_PI * earth_rad * mk + 1) : 1.;
-		t1 = earth_curvature * (twopi * NEWTON_G);
-		if (Ctrl->F.mode == GRAVFFT_FAA)
-			t1 *= 1.0e5;     /* to have it in mGals */
-		else                 /* Must be the GEOID case */
-			t1 /= (NORMAL_GRAVITY * mk * twopi);
-		t2 = exp(-twopi * mk * Ctrl->misc.z_level) - exp(-twopi * mk * Ctrl->Z.zm) / (1 + alfa*pow(mk,4.));
-		datac[k] += (gmt_grdfloat) ((Ctrl->T.rho_cw * t1 * t2) * t / f * raised[k]);
-		datac[k+1] += (gmt_grdfloat) ((Ctrl->T.rho_cw * t1 * t2) * t / f * raised[k+1]);
+		t1 = earth_curvature * (TWO_PI * NEWTON_G);
+		t2 = exp(-TWO_PI * mk * Ctrl->misc.z_level) - exp(-TWO_PI * mk * Ctrl->Z.zm) / (1 + alfa*pow(mk,4.));
+
+		/* The switch below is probably only correct for FAA and possibly GEOID.
+		 * THose where the only two cases covered by JL's code */
+
+		switch (Ctrl->F.mode) {
+			case GRAVFFT_FAA:
+				t1 *= 1.0e5;     /* To have it in mGals */
+				datac[k]   += (gmt_grdfloat) ((Ctrl->T.rho_cw * t1 * t2) * t / f * raised[k]);
+				datac[k+1] += (gmt_grdfloat) ((Ctrl->T.rho_cw * t1 * t2) * t / f * raised[k+1]);
+				break;
+			case GRAVFFT_GEOID:
+				if (mk > 0.0) t1 /= (NORMAL_GRAVITY * mk * TWO_PI);
```

Comment:
This ``if (mk > 0.0)`` condition was not in the old code, which was probably a bug.